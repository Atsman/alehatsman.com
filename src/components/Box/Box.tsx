import styled from '@emotion/styled'
import {
  space,
  layout,
  color,
  flexbox,
  border,
  typography
} from 'styled-system'

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    minWidth: 0
  },
  border,
  color,
  flexbox,
  layout,
  space,
  typography
)

export const Flex = styled(Box)({
  display: 'flex'
})
