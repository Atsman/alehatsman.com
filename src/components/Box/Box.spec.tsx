import React from 'react'
import { render } from '@testing-library/react'
import { Box, Flex } from './Box'

describe('Box', () => {
  it('should match snapshot', () => {
    expect(render(<Box />).asFragment()).toMatchSnapshot()
  })
})

describe('Flex', () => {
  it('should match snapshot', () => {
    expect(render(<Flex />).asFragment()).toMatchSnapshot()
  })
})
