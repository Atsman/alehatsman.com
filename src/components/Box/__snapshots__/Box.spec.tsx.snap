// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  min-width: 0;
}

<div
    class="emotion-0 emotion-1"
  />
</DocumentFragment>
`;

exports[`Flex should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="emotion-0 emotion-1"
  />
</DocumentFragment>
`;
